addons/binding/org.openhab.binding.autelis/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,568 @@
+		<description>The current state of feature/macro 3</description>
+		<state readOnly="false">
+		</state>
+	</channel-type>
+	<channel-type id="equipment-feature4">
+		<item-type>Number</item-type>
+		<label>Feature 4</label>
+		<description>The current state of feature/macro 4</description>
+		<state readOnly="false">
+		</state>
+	</channel-type>
+	<channel-type id="equipment-feature5">
+		<item-type>Number</item-type>
+		<label>Feature 5</label>
+		<description>The current state of feature/macro 5</description>
+		<state readOnly="false">
+		</state>
+	</channel-type>
+	<channel-type id="equipment-feature6">
+		<item-type>Number</item-type>
+		<label>Feature 6</label>
+		<description>The current state of feature/macro 6</description>
+		<state readOnly="false">
+		</state>
+	</channel-type>
+	<channel-type id="equipment-feature7">
+		<item-type>Number</item-type>
+		<label>Feature 7</label>
+		<description>The current state of feature/macro 7</description>
+		<state readOnly="false">
+		</state>
+	</channel-type>
+	<channel-type id="equipment-feature8">
+		<item-type>Number</item-type>
+		<label>Feature 8</label>
+		<description>The current state of feature/macro 8</description>
+		<state readOnly="false">
+		</state>
+	</channel-type>
+	<channel-type id="equipment-feature9">
+		<item-type>Number</item-type>
+		<label>Feature 9</label>
+		<description>The current state of feature/macro 9</description>
+		<state readOnly="false">
+		</state>
+	</channel-type>
+	<channel-type id="equipment-feature10">
+		<item-type>Number</item-type>
+		<label>Feature 10</label>
+		<description>The current state of feature/macro 10</description>
+		<state readOnly="false">
+		</state>
+	</channel-type>
+	<channel-type id="temp-poolht">
+		<item-type>Number</item-type>
+		<label>Pool Heater Status</label>
+		<description>The current pool/hi-temp heater settings</description>
+		<state readOnly="false">
+			<options>
+				<option value="0">Off</option>
+				<option value="1">Heater</option>
+				<option value="2">Solar Preferred</option>
+				<option value="3">Solar Only</option>
+			</options>
+		</state>
+	</channel-type>
+	<channel-type id="temp-spaht">
+		<item-type>Switch</item-type>
+		<label>The current spa/lo-temp heater settings</label>
+		<description>Spa Heat</description>
+		<state readOnly="false">
+			<options>
+				<option value="0">Off</option>
+				<option value="1">Heater</option>
+				<option value="2">Solar Preferred</option>
+				<option value="3">Solar Only</option>
+			</options>
+		</state>
+	</channel-type>
+	<channel-type id="temp-htstatus">
+		<item-type>Number</item-type>
+		<label>Heat Status</label>
+		<description>Heat Status </description>
+		<state readOnly="true">
+		</state>
+	</channel-type>
+	<channel-type id="temp-poolsp">
+		<item-type>Number</item-type>
+		<label>Pool Heat Setpoint</label>
+		<description>The current pool/hi-temp setpoint</description>
+	</channel-type>
+	<channel-type id="temp-spasp">
```

Comment:
Very good question, I was afraid that you would ask ;-)
I do not have a good answer - I think it would be probably best if the thing would dynamically adapt its channels when the mode is changed between C/F - unfortunately I noticed that it is not possible to provide a state description with a dynamically created channel (@dnobel, did you know?). So right now, maybe I have to revoke my wish and we better go without a pattern and min/max here...
The other option would be that you have always two channels, one for F and one for C; even though your controller offers only one of these values, you could potentially automatically convert it. But I am not sure if this is worth the effort.